balancer-js/src/modules/pricing/priceImpact.ts

```diff
@@ -14,12 +14,14 @@ function calcPriceImpactJoin(
   bptZeroPriceImpact: bigint
 ): bigint {
   // 1 - (bptAmount/bptZeroPI)
-  return ONE - SolidityMaths.divDownFixed(bptAmount, bptZeroPriceImpact);
+  const pi = ONE - SolidityMaths.divDownFixed(bptAmount, bptZeroPriceImpact);
+  return pi < 0 ? BigInt(0) : pi;
```

Comment:
Is there a scenario where we're seeing negative price impact being calculated? I'm wondering if we should throw an error rather than forcing a zero amount in that case.

Reply:
Yeah. There are cases where it returns a small negative amount instead of zero. Sergio checked this out and his opinion was its was negligible and ok to return 0.